cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMakerCluster.java

```diff
@@ -0,0 +1,405 @@
+        return cluster + KafkaMirrorMakerCluster.NAME_SUFFIX;
+    }
+
+    public static String serviceName(String cluster) {
+        return cluster + KafkaMirrorMakerCluster.SERVICE_NAME_SUFFIX;
+    }
+
+    public static String logAndMetricsConfigName(String cluster) {
+        return cluster + KafkaMirrorMakerCluster.METRICS_AND_LOG_CONFIG_SUFFIX;
+    }
+
+    public static KafkaMirrorMakerCluster fromCrd(KafkaMirrorMaker kafkaMirrorMaker) {
+        KafkaMirrorMakerCluster kafkaMirrorMakerCluster = new KafkaMirrorMakerCluster(kafkaMirrorMaker.getMetadata().getNamespace(),
+                kafkaMirrorMaker.getMetadata().getName(),
+                Labels.fromResource(kafkaMirrorMaker).withKind(kafkaMirrorMaker.getKind()));
+
+
+        kafkaMirrorMakerCluster.setReplicas(kafkaMirrorMaker.getSpec() != null && kafkaMirrorMaker.getSpec().getReplicas() > 0 ? kafkaMirrorMaker.getSpec().getReplicas() : DEFAULT_REPLICAS);
+
+        if (kafkaMirrorMaker.getSpec() != null) {
+            kafkaMirrorMakerCluster.setWhitelist(kafkaMirrorMaker.getSpec().getWhitelist());
+            kafkaMirrorMakerCluster.setProducer(kafkaMirrorMaker.getSpec().getProducer());
+            kafkaMirrorMakerCluster.setConsumer(kafkaMirrorMaker.getSpec().getConsumer());
+
+            Map<String, Object> metrics = kafkaMirrorMaker.getSpec().getMetrics();
+            if (metrics != null && !metrics.isEmpty()) {
+                kafkaMirrorMakerCluster.setMetricsEnabled(true);
+                kafkaMirrorMakerCluster.setMetricsConfig(metrics.entrySet());
+            }
+        }
+
+        return kafkaMirrorMakerCluster;
+    }
+
+    /**
+     * Abstracts the calling of setters on a (subclass of) KafkaMirrorMakerCluster
+     * from the instantiation of the (subclass of) KafkaMirrorMakerCluster,
+     * thus permitting reuse of the setter-calling code for subclasses.
+     */
+    protected static <C extends KafkaMirrorMakerCluster> C fromSpec(KafkaMirrorMakerSpec spec, C kafkaMirrorMaker) {
+        kafkaMirrorMaker.setReplicas(spec != null && spec.getReplicas() > 0 ? spec.getReplicas() : DEFAULT_REPLICAS);
+
+        if (spec != null) {
+            kafkaMirrorMaker.setWhitelist(spec.getWhitelist());
+            kafkaMirrorMaker.setProducer(spec.getProducer());
+            kafkaMirrorMaker.setConsumer(spec.getConsumer());
+
+            Map<String, Object> metrics = spec.getMetrics();
+            if (metrics != null && !metrics.isEmpty()) {
+                kafkaMirrorMaker.setMetricsEnabled(true);
+                kafkaMirrorMaker.setMetricsConfig(metrics.entrySet());
+            }
+        }
+        return kafkaMirrorMaker;
+    }
+
+    public Service generateService() {
+        List<ServicePort> ports = new ArrayList<>(1);
+        if (isMetricsEnabled()) {
+            ports.add(createServicePort(METRICS_PORT_NAME, METRICS_PORT, METRICS_PORT, "TCP"));
+        }
+
+        return createService("ClusterIP", ports, getPrometheusAnnotations());
```

Comment:
I wonder if we really need the service just for metrics. But maybe we should just return null here when metrics are not enabled (=no port will be open on the service)?